salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- audit devices
+        """
+        log.info("Loading audit devices from local config...")
+        devices = []
+        if configs:
+            try:
+                for audit_device in configs:
+                    config = None
+                    if 'config' in audit_device:
+                        if audit_device['config']:
+                            config = OrderedDict(
+                                sorted(audit_device["config"].items()))
+
+                    devices.append(
+                        VaultAuditDevice(
+                            type=audit_device["type"],
+                            path=audit_device["path"],
+                            description=audit_device["description"],
+                            config=config
+                        )
+                    )
+            except Exception:
+                raise
+
+        log.info('Finished loading audit devices from local config.')
+
+        return devices
+
+    def configure_audit_devices(self, client, remote_devices, local_devices, ret):
+        """Compare and configure audit devices
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_devices {list} -- audit devices from remote vault server
+            local_devices {list} -- audit devices from local vault config file
+            ret {dict} -- salt state result
+        """
+        log.info('Processing and configuring audit devices...')
+        new_audit_devices = []
+        try:
+            for audit_device in local_devices:
+                log.debug('Checking if audit device "%s" at path "%s" is enabled...',
+                          audit_device.type, audit_device.path)
+
+                if audit_device in remote_devices:
+                    log.debug('Audit device "%s" at path "%s" is already enabled.',
+                              audit_device.type, audit_device.path)
+                else:
+                    log.debug(
+                        'Audit device "%s" at path "%s" is not enabled. Enabling now...',
+                        audit_device.type,
+                        audit_device.path
+                    )
+                    new_audit_devices.append(audit_device.type)
+                    client.sys.enable_audit_device(
+                        device_type=audit_device.type,
+                        path=audit_device.path,
+                        description=audit_device.description,
+                        options=audit_device.config
+                    )
+                    log.debug('Audit device "%s" at path "%s" is enabled.',
+                              audit_device.type, audit_device.path)
+
+            # Build return object
+            ret['changes']['old'] = [ob.type for ob in remote_devices]
+            ret['changes']['new'] = new_audit_devices or "No changes"
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.